src/test/compile-fail/union-empty.rs

```diff
@@ -0,0 +1,15 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(untagged_unions)]
+
+union U {} //~ ERROR unions cannot have zero fields
```

Comment:
The semantics is unclear, is `union U {}` an uninhabited type like `enum E {}` or an empty type like `struct E {}`.
Both can be implemented by either keeping the "union expressions should have exactly one field" rule or relaxing it to "union expressions should have at most one field".